code/modules/hydroponics/unique_plant_genes.dm

```diff
@@ -14,26 +14,20 @@
 		return
 	var/obj/item/seeds/our_seed = our_plant.get_plant_seed()
 	shield_uses = round(our_seed.potency / 20)
-	our_plant.AddComponent(/datum/component/anti_magic, TRUE, TRUE, FALSE, ITEM_SLOT_HANDS, shield_uses, TRUE, CALLBACK(src, .proc/block_magic), CALLBACK(src, .proc/expire)) //deliver us from evil o melon god
-
-/*
- * The proc called when the holymelon successfully blocks a spell.
- *
- * user - the mob who's using the melon
- * major - whether the spell was 'major' or not
- * our_plant - our plant, who's eating the magic spell
- */
-/datum/plant_gene/trait/anti_magic/proc/block_magic(mob/user, major, obj/item/our_plant)
-	if(major)
-		to_chat(user, "<span class='warning'>[our_plant] hums slightly, and seems to decay a bit.</span>")
-
-/*
- * The proc called when the holymelon uses up all of its anti-magic charges.
- *
- * user - the mob who's using the melon
- * major - whether the spell was 'major' or not
- * our_plant - our plant, who tragically melted protecting us from magics
- */
+	//deliver us from evil o melon god
+	our_plant.AddComponent(/datum/component/anti_magic, \
+		antimagic_flags = MAGIC_RESISTANCE|MAGIC_RESISTANCE_HOLY, \
+		inventory_flags = ITEM_SLOT_HANDS, \
+		charges = shield_uses, \
+		drain_antimagic = CALLBACK(src, .proc/drain_antimagic), \
+		expiration = CALLBACK(src, .proc/expire), \
+	)
+
+/// When the antimagic plant gene is drained a anti-magic charge.
+/datum/plant_gene/trait/anti_magic/proc/drain_antimagic(mob/user, obj/item/our_plant)
+	to_chat(user, "<span class='warning'>[our_plant] hums slightly, and seems to decay a bit.</span>")
+
+/// When the antimagic plant gene uses up all of its anti-magic charges.
```

Comment:
```suggestion
/// When the plant our gene is hosted in is drained of all of its anti-magic charges.
```
Bonus points if you can replace the old <span> use in this proc with a span macro

Reply:
Updated to_chat with https://github.com/tgstation/tgstation/pull/64124/commits/4326f81c9a9b08b8dc0a0d1c78a9dd493c340e1a